### PR TITLE
Wired up e2e tests to use local KIND cluster

### PR DESCRIPTION
**Pull request by rinormaloku**
This will resolve the issue #62.

The steps have to be executed one by one, as shown below:
```
make create-kindcluster
make install-istio
make install-smi-crds
make test-e2e
```

Which makes it very simple to integrate with CI platforms.

### CODE REVIEW

**Comment by michelleN** [comment]
Thanks @rinormaloku ! This looks great. I'm running into an error when running `make create-kindcluster`: 
```console
smi-adapter-istio [rinormaloku-kind-cluster*]$ make create-kindcluster
installing kind
GO111MODULE="on" go get sigs.k8s.io/kind@v0.5.1
go: github.com/peterbourgon/diskv@v0.0.0-20190425080511-0be1b92a6df0: go.mod has post-v0 module path "github.com/peterbourgon/diskv/v3" at revision 0be1b92a6df0
go: finding k8s.io/kubernetes v1.11.7-beta.0.0.20181219023948-b875d52ea96d
go: finding k8s.io/api v0.0.0-20180904230853-4e7be11eab3f
go: finding k8s.io/client-go v0.0.0-20180718001006-59698c7d9724
go: finding k8s.io/apiextensions-apiserver v0.0.0-20180905004947-16750353bf97
go: finding k8s.io/apimachinery v0.0.0-20180904193909-def12e63c512
go: error loading module requirements
make: *** [create-kindcluster] Error 1
```

Lmk if u have any ideas about why I might be getting this error.

**Comment by stefanprodan** [comment]
Instead of compiling kind in CI (it takes about 2m) I've been using the binary https://github.com/weaveworks/flagger/blob/master/test/e2e-kind.sh#L19

File: Makefile [hunk 1]
```diff
@@ -63,11 +66,33 @@ test-unit:
...
 	go get -u github.com/golang/dep/cmd/dep
 endif
 	dep ensure
+
+create-kindcluster:
+ifndef HAS_KIND
+	@echo "installing kind"
+	GO111MODULE="on" go get sigs.k8s.io/kind@v0.5.1
+endif
+	kind create cluster --name local-kind  --image kindest/node:v1.15.3
```
Comment (michelleN) [inline Makefile]: should we add `export KIND_KUBECONFIG="$(kind get kubeconfig-path --name="local-kind")"` here? `KIND_KUBECONFIG` was not set for me in the next step.

**Comment by rinormaloku** [comment]
@michelleN will check the issue in the coming days. 
And thanks @stefanprodan for the great suggestion!

File: Makefile (hunk 1 above)
Comment (rinormaloku) [reply Makefile]: @michelleN when we need to set the KUBECONFIG this command has to be used:
```
KIND_KUBECONFIG = $(shell kind get kubeconfig-path --name=local-kind)
```
as shown here:
```
install-smi-crds:
	kubectl apply -f https://raw.githubusercontent.com/deislabs/smi-adapter-istio/master/deploy/crds/crds.yaml \
		--kubeconfig=$(KIND_KUBECONFIG)
```

**Comment by rinormaloku** [comment]
> Thanks @rinormaloku ! This looks great. I'm running into an error when running `make create-kindcluster`:
> 
> ```
> smi-adapter-istio [rinormaloku-kind-cluster*]$ make create-kindcluster
> installing kind
> GO111MODULE="on" go get sigs.k8s.io/kind@v0.5.1
> go: github.com/peterbourgon/diskv@v0.0.0-20190425080511-0be1b92a6df0: go.mod has post-v0 module path "github.com/peterbourgon/diskv/v3" at revision 0be1b92a6df0
> go: finding k8s.io/kubernetes v1.11.7-beta.0.0.20181219023948-b875d52ea96d
> go: finding k8s.io/api v0.0.0-20180904230853-4e7be11eab3f
> go: finding k8s.io/client-go v0.0.0-20180718001006-59698c7d9724
> go: finding k8s.io/apiextensions-apiserver v0.0.0-20180905004947-16750353bf97
> go: finding k8s.io/apimachinery v0.0.0-20180904193909-def12e63c512
> go: error loading module requirements
> make: *** [create-kindcluster] Error 1
> ```
> 
> Lmk if u have any ideas about why I might be getting this error. I'll go ahead and just install the binary from the official releases for now to get past this.

I fixed it using the workaround suggested by @stefanprodan, where instead of building the binary we download the prebuilt one.

**Pull request merged**